packages/presentation/lib/app/app_bloc.dart

```diff
@@ -104,6 +112,11 @@ class _AppBloc extends BlocImpl implements AppBloc {
   BasePage? _currentPage() => _appData.pages.lastOrNull;
 
   void _updateData() {
+    final screenLog = FirebaseAnalyticsModel(
+      eventName: EventName.screenView,
+      eventScreenLog: {EventName.screen: _currentPage()?.name},
+    );
+    analyticsUseCase(screenLog);
```

Comment:
up to u, but creating a FirebaseAnaliticsModel, for every button/screen makes code bulky. U can create two use case for analytics (screen and button) and move part of this logic there. But repeat, up to u =)

Reply:
if you touch patterns, then yes, you can break, but as for me, one or two arguments, the weather will not play and there will be more code